commands/assets/add.go

```diff
@@ -0,0 +1,135 @@
+package assets
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+
+	"github.com/netlify/netlifyctl/commands/middleware"
+	"github.com/netlify/netlifyctl/context"
+	"github.com/netlify/open-api/go/plumbing/operations"
+	"github.com/spf13/cobra"
+)
+
+type assetsAddCmd struct {
+	private bool
+}
+
+func setupAddCommand(middlewares []middleware.Middleware) *cobra.Command {
+	cmd := &assetsAddCmd{}
+	ccmd := &cobra.Command{
+		Use:   "add [ASSET PATH 1] [ASSET PATH 2] ...",
+		Short: "Add an asset to a site",
+		Long:  "Add an asset to a site",
+	}
+	ccmd.Flags().BoolVarP(&cmd.private, "private", "p", false, "make the asset private")
+
+	return middleware.SetupCommand(ccmd, cmd.AddAsset, middlewares)
+}
+
+func (c *assetsAddCmd) AddAsset(ctx context.Context, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("missing asset paths to upload")
+	}
+
+	siteId, err := siteIdForCommand(cmd)
+	if err != nil {
+		return err
+	}
+
+	dups := make(map[string]os.FileInfo)
+	for _, arg := range args {
+		fi, err := os.Stat(arg)
+		if err != nil {
+			return fmt.Errorf("unable to access asset with path %s, all assets must exist: %v", arg, err)
+		}
+
+		if _, dup := dups[arg]; dup {
+			fmt.Printf("[WARNING] duplicated asset %s, will be uploaded only once\n", arg)
+		}
+
+		dups[arg] = fi
+	}
+
+	client := context.GetClient(ctx)
+	params := operations.NewCreateSiteAssetParams().WithSiteID(siteId)
+	if c.private {
+		visibility := "private"
+		params = params.WithVisibility(&visibility)
+	}
+
+	for fp, fi := range dups {
+		body, err := os.Open(fp)
+		if err != nil {
+			return err
```

Comment:
might be good to give more informed error here - like check if the file moved or is a dir.